pkg/services/ngalert/metrics/ngalert.go

```diff
@@ -52,6 +52,7 @@ type Scheduler struct {
 	EvalFailures             *prometheus.CounterVec
 	EvalDuration             *prometheus.SummaryVec
 	GetAlertRulesDuration    prometheus.Histogram
+	GetAlertRulesFailures    prometheus.Counter
```

Comment:
I don't think a counter tells us much - I feel strongly about a gauge with the hash of the current rule set to track changes over time.

Something like `RuleSetHash` for the variable name and its type should be a `prometheus.Gauge`.

For the metric details:

I suggest the name `shedule_ruleset_hash", where its value is the hash -> float64. If you're short on time, feel free to leave it for the next PR.

Reply:
I started to work on this but one observation I have is that the values will be so large (64 bit checksum as a float64) that I think it will be difficult to show in Grafana because the Y axis will have to scale between min and max of a 64 bit float?